modules/investigating-kernel-crashes.adoc

```diff
@@ -0,0 +1,95 @@
+
+{op-system} ships with `kexec-tools`, but manual configuration is required to enable `kdump`.
+
+The following are the basic steps needed to enable `kdump` on {op-system}.
+
+.  Memory must be reserved for the crash kernel during booting of the first kernel. Kernel arguments can be provided like this:
++
+[source, terminal]
+----
+sudo rpm-ostree kargs --append='crashkernel=256M'
+----
+
+.  By default, the path in which the vmcore will be saved is `/var/crash`. It is also possible to write the dump over the network or to some other location on the local system by editing `/etc/kdump.conf`. For additional information, see `kdump.conf(5)` and the comments in `/etc/kdump.conf` and `/etc/sysconfig/kdump`.
+ifdef::openshift-enterprise[]
+Also refer to the RHEL `kdump` documentation link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/system_design_guide/installing-and-configuring-kdump_system-design-guide#configuring-the-kdump-target_configuring-kdump-on-the-command-line[here] for further information on configuring the dump target.
+endif::[]
+
+ifdef::openshift-enterprise[]
+.  Because `kdump` has trouble finding the correct bootimage location on {op-system}, the `KDUMP_BOOTDIR` variable must be manually set in `/etc/kdump.conf`. You can use `/proc/cmdline` to figure out the `ostree` boot location. For example:
++
+[source, terminal]
+----
+BOOT_LOC=/boot$(cat /proc/cmdline | egrep -o "/ostree/.*/vmlinuz" | sed -e "s|/vmlinuz||g")
+sudo sed -i "s|^#KDUMP_BOOTDIR=\"/boot\"|KDUMP_BOOTDIR=\"${BOOT_LOC}\"|" /etc/sysconfig/kdump
+----
+
+.  Configure `/etc/sysconfig/kdump` to not use the default link:https://man7.org/linux/man-pages/man2/kexec_file_load.2.html[file-based `kexec` syscall] for loading the crash kernel.
++
+[source, terminal]
+----
+sudo sed -i "s|^KEXEC_ARGS=\"-s\"|KEXEC_ARGS=\"\"|" /etc/sysconfig/kdump
+----
+endif::[]
+
+.  Enable the `kdump` systemd service.
++
+[source, terminal]
+----
+sudo systemctl enable kdump.service
+----
+
+.  Reboot your system.
++
+[source, terminal]
+----
+sudo systemctl reboot
+----
+
+.  Ensure that `kdump` has loaded a crash kernel by checking that `kdump.service` has started and exited successfully and that `cat /sys/kernel/kexec_crash_loaded` prints `1`.
+
+=== Enabling kdump on day-1
+`kdump` is intended to be enabled per-node to debug kernel problems; it is not recommended to enable `kdump` on all of your nodes in the cluster. Nevertheless, although machine-specific `MachineConfigs` are not yet supported, it is possible to do the above through a systemd unit in a `MachineConfig` object on day-1 and have kdump enabled on all nodes in the cluster. You can create a `MachineConfig` object and inject that object into the set of manifest files used by Ignition during cluster setup. Please see xref:../../installing/install_config/installing-customizing.adoc["Customizing nodes"] for more information and examples on how to use Ignition configs.
```

Comment:
s/problems; it/problems. It 

s/Nevertheless, although/Although 

s/Please see xref:../../installing/install_config/installing-customizing.adoc["Customizing nodes"]/See xref:../../installing/install_config/installing-customizing.adoc[Customizing nodes]

Reply:
Follow-up we may need to remove this xref/build it in the assembly because we have some strict rules for linking by xrefs in modules.

Reply:
As Brandi mentioned, no xrefs are allowed in modules, so I suggest:

> See "Customizing nodes" in the _Installing --> Installation configuration_ section for more information and examples of how to use Ignition configs.